exchanges/okx/okx_websocket.go

```diff
@@ -282,56 +275,17 @@ func (ok *Okx) WsAuth(ctx context.Context, dialer *websocket.Dialer) error {
 		return err
 	}
 	base64Sign := crypto.Base64Encode(hmac)
-	req := WebsocketEventRequest{
-		Operation: operationLogin,
-		Arguments: []WebsocketLoginData{
-			{
-				APIKey:     creds.Key,
-				Passphrase: creds.ClientID,
-				Timestamp:  timeUnix,
-				Sign:       base64Sign,
-			},
+	args := []WebsocketLoginData{
+		{
+			APIKey:     creds.Key,
+			Passphrase: creds.ClientID,
+			Timestamp:  timeUnix,
+			Sign:       base64Sign,
 		},
 	}
-	err = ok.Websocket.AuthConn.SendJSONMessage(ctx, request.Unset, req)
-	if err != nil {
-		return err
-	}
-	timer := time.NewTimer(ok.WebsocketResponseCheckTimeout)
-	randomID, err := common.GenerateRandomString(16)
-	if err != nil {
-		return fmt.Errorf("%w, generating random string for incoming websocket response failed", err)
-	}
-	wsResponse := make(chan *wsIncomingData)
-	ok.WsResponseMultiplexer.Register <- &wsRequestInfo{
-		ID:    randomID,
-		Chan:  wsResponse,
-		Event: operationLogin,
-	}
-	ok.WsRequestSemaphore <- 1
-	defer func() {
-		<-ok.WsRequestSemaphore
-	}()
-	defer func() { ok.WsResponseMultiplexer.Unregister <- randomID }()
-	for {
-		select {
-		case data := <-wsResponse:
-			if data.Event == operationLogin && data.Code == "0" {
-				ok.Websocket.SetCanUseAuthenticatedEndpoints(true)
-				return nil
-			} else if data.Event == "error" &&
-				(data.Code == "60022" || data.Code == "60009") {
-				ok.Websocket.SetCanUseAuthenticatedEndpoints(false)
-				return fmt.Errorf("authentication failed with error: %v", ErrorCodes[data.Code])
-			}
-			continue
-		case <-timer.C:
-			timer.Stop()
-			return fmt.Errorf("%s websocket connection: timeout waiting for response with an operation: %v",
-				ok.Name,
-				req.Operation)
-		}
-	}
+	var nothing interface{}
+	err = ok.SendAuthenticatedWebsocketRequest(ctx, "login-response", operationLogin, args, &nothing)
```

Comment:
```suggestion
	err = ok.SendAuthenticatedWebsocketRequest(ctx, "login-response", operationLogin, args, nil)
```

Remove the `response` `nil` check from `SendAuthenticatedWebsocketRequest` and you won't need to jank it up